pwndbg/commands/leakfind.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Find a chain of leaks given some starting address.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+from queue import *
+
+import gdb
+
+import pwndbg.color.chain as C
+import pwndbg.color.memory as M
+import pwndbg.color.theme as theme
+import pwndbg.commands
+import pwndbg.vmmap
+from pwndbg.chain import config_arrow_right
+
+
+# Used to recursively print the pointer chain. 
+# addr is a pointer. It is taken to be a child pointer.
+# visited_map is a map of children -> (parent,parent_start)
+def get_rec_addr_string(addr,visited_map):
+    page = pwndbg.vmmap.find(addr)
+    arrow_right = C.arrow(' %s ' % config_arrow_right)
+
+    if not page is None:
+        if not addr in visited_map:
+            return ""
+        
+        parent_info = visited_map[addr]
+        parent = parent_info[0]
+        parent_base_addr = parent_info[1]
+        if parent - parent_base_addr < 0:
+            curText = hex(parent_base_addr) + hex(parent - parent_base_addr)
+        else:
+            curText = hex(parent_base_addr) + "+" + hex(parent - parent_base_addr)
+        if parent_base_addr == addr:
+            return ""
+        return get_rec_addr_string(parent_base_addr,visited_map) + M.get(parent_base_addr,text=curText) + arrow_right
+    else:
+        return ""
+
+
+# Useful for debugging. Prints a map of child -> (parent, parent_start)
+def dbg_print_map(maps):
+    for child, parent_info in maps.items():
+        print("0x%x + (0x%x, 0x%x)" % (child, parent_info[0], parent_info[1]))
+
+parser = argparse.ArgumentParser()
+parser.description = """
+Attempt to find a leak chain given a starting address. Scans memory near the given address, looks for pointers, 
+and continues that process to attempt to find leaks.\n
+Example: leakfind $rsp --page_name=filename --max_offset=0x48 --max_depth=6. This would look for any chains of leaks that point to a section in filename which begin near $rsp, are never 0x48 bytes further from a known pointer, 
+and are a maximum length of 6\n
+"""
+parser.add_argument("address",help="Starting address to find a leak chain from")
+parser.add_argument("-p", "--page_name", type=str, nargs="?", default=None, help="Substring required to be part of the name of any found pages")
+parser.add_argument("-o", "--max_offset", default=0x48, nargs="?", help="Max offset to add to addresses when looking for leak")
+parser.add_argument("-d", "--max_depth", default=0x4, nargs="?", help="Maximum depth to follow pointers to")
+parser.add_argument("-s", "--stride", nargs="?", default=0x1, help="Stride to add to add between pointers considered. For example, if this is 4 it would only consider pointers at an offset divisible by 4 from the starting pointer")
```

Comment:
```suggestion
parser.add_argument("-s", "--step", nargs="?", default=0x1, help="Step to add between pointers so they are considered. For example, if this is 4 it would only consider pointers at an offset divisible by 4 from the starting pointer")
```